avoid deprecated 'gtk_menu_popup'

sc0w opened the pull request.

raveit65 commented on pluma/pluma-window.c:
```diff
@@ -3442,9 +3442,7 @@ show_notebook_popup_menu (GtkNotebook    *notebook,
-		gtk_menu_popup (GTK_MENU (menu), NULL, NULL,
-				NULL, NULL,
-				event->button, event->time);
+		gtk_menu_popup_at_pointer (GTK_MENU (menu), NULL);
```
Why using different functions here?
Later you choose gtk_menu_popup_at_widget as replacement?

sc0w replied:
here:

press the secondary mouse button in the tab:

![pluma-window1](https://user-images.githubusercontent.com/7734191/40308792-9e30a1dc-5d07-11e8-8519-339a1adfe78d.png)

later:

get the tab focus with [ctrl]+[shift]+[tab] and press the "properties" key in your keyboard:

![pluma-window2](https://user-images.githubusercontent.com/7734191/40308814-b3cb664e-5d07-11e8-8414-ea5c59beb86b.png)

the same behavior without this PR

raveit65 replied:
Ok, but isn't it more elegant to use gtk_menu_popup_at_widget for pointer too, if possible?
In result the menu opens always at the same place and the document title is always readable.
But it is OK for me if you don't like that....

sc0w replied:
In the PRs of deprecated functions, I like to respect the original behavior

But I have no problem if you decide to change it in future PRs, I agree with your idea

vkareh replied:
I would love to see this fixed too, but yeah this can be a new PR.

raveit65 commented:
How can i execute the show-line-numbers menu ?

raveit65 commented:
With PR:
- Menus on notebook tabs do what code promised.
- menu_popup_at_treeview_selection fixes position error of menu in file browser sidebar, and open fine with pointer and keyboard.
- Menu in document panel opens fine with pointer and keyboard
- But i can't close bth menu in sidebar with esc key!
- Combobox menu in statusbar opens fine with mouse but not with keyboard, but this is already broken in master. So no new issue.
- Menu in pluma-window opens fine with pointer and keyboard
- line-numbers menu open s with mouse but 'reponse issue' from master still exists.

sc0w commented:
@lukefromdc 

> I found another problem with NULL in gtk_menu_popup_at_pointer in this, causing poor response from the context menu in the print preview.

The changes in `pluma-print-preview.c` affects this menu popup in print preview (primary mouse button):

![print_preview_popup](https://user-images.githubusercontent.com/7734191/40322199-e2fd24e6-5d31-11e8-9e03-afcaeff07a09.png)

It works fine here, probably you have the issue https://github.com/mate-desktop/pluma/issues/321

The menu popup...

![menu_popup_print](https://user-images.githubusercontent.com/7734191/40312979-9b958ab6-5d14-11e8-9183-d4acb53ce0ac.png)

..only must see in the tab header, if not, that is a bug, you can compare the same behavior in git master

> pluma_utils_menu_position_under_tree_view does not seem to be used within Pluma itself.

I will take a look later

@raveit65 

> But i can't close both menus in sidebar with esc key!

I can

> line-numbers menu open s with mouse but 'reponse issue' from master still exists.

Problems with line numbers menu popup?

It was fixed in https://github.com/mate-desktop/pluma/commit/a28da51ba62baded1ea23e6ae9f4c907d443b242

problems in git master? are you sure?

I confirm the problem in pluma 1.20.1, I cherry picked the fix to the 1.20 branch

It works fine for me in git master and this PR

![line_numbers](https://user-images.githubusercontent.com/7734191/40322823-e25d61fc-5d33-11e8-80a8-73f2c200ee31.png)

raveit65 commented:
>> But i can't close both menus in sidebar with esc key!

> I can

weird, i tested it again an now it works.

> problems in git master? are you sure?

I tested it before i applied commits from this PR.

sc0w commented:
@raveit65 ok, please open new issue with detailed information

@lukefromdc line numbers menu popup works fine for you?

lukefromdc commented:
I did not find any problem with the line number popup

raveit65 commented:
With or without the PR with the first try the menu opens fine, but if i want to open the menu again i need to click 2 times.
Tested with 1.18.4 and master+PR.
In other PRs we fixed that with don't using `NULL`
Well, for normal usage you will never see this menu as it displays only if you have empty space there.
(a right-click on a existing number doesn't work)
So ignore it....

sc0w commented:
I confirm the "two times click" issue in 1.18.4, it was fixed later in https://github.com/mate-desktop/pluma/commit/618b640af64da8e4bb90ac98e5376fed670c65d6 in the 1.18 branch

Confirmed in 1.20.1, and fixed later with https://github.com/mate-desktop/pluma/commit/d7a6b5096735d6fc5c8825cdf0f3da4d2155f287 in the 1.20 branch

And fixed in master with https://github.com/mate-desktop/pluma/commit/a28da51ba62baded1ea23e6ae9f4c907d443b242 in the PR https://github.com/mate-desktop/pluma/pull/312

I don't understand why you have the issue if it was already fixed

raveit65 commented:
> I don't understand why you have the issue if it was already fixed

Why?
1. I am not a machine and sometimes i missed a patch and the patch isn't included my rpm, especial in repos where every day comes a new commit ;) 
2. It wasn't clear said that https://github.com/mate-desktop/pluma/pull/312 fixes the problem.

Using the patch fixes the click-response issue, but clicking on line numbers don't work in non-empty documents. (same problem in master)

sc0w commented:
> Using the patch fixes the click-response issue, but clicking on line numbers don't work. (same problem in master)

yes, I think "click on line numbers" never worked in the past to show the checkmenuitem

lukefromdc commented:
OK, 
https://github.com/mate-desktop/pluma/pull/324 works, I had thought that popup menu was supposed to appear since it did some of the time and I had not noticed it until testing this.  That in turns voids the issue I found in testing this PR, since that popup was not supposed to appear at all anyway

sc0w commented:
@lukefromdc 

> pluma_utils_menu_position_under_tree_view does not seem to be used within Pluma itself. I was able to remove it from pluma-utils.c and pluma-utils.h with no change in behavior. 

yes, confirmed, done in latest commit

raveit65 commented:
> I would love to see this fixed too, but yeah this can be a new PR.

https://github.com/mate-desktop/pluma/pull/325
Here we go.
If i did something wrong please commit directly on PR.
